wurst/systems/spells/gatherer/TeleGathering.wurst

```diff
@@ -0,0 +1,161 @@
+let ABILITY_ID_OMNI_TELEGATHER = 'A06V'
+let ABILITY_ID_CASTER_BUFF = 'A056'
+let CASTER_BUFF_ID = 'B00H'
+let TARGET_BUFF_ID = 'B00I'
+let SORT_DISTANCE_INCREMENT = 150.
+let SORT_ANGLE_INCREMENT = 45
+
+HashMap<unit, TeleGatherInstance> instances = new HashMap<unit, TeleGatherInstance>()
+class SortingPosition
+    vec2 pos
+    construct(vec2 pos)
+        this.pos = pos
+
+class TeleGatherInstance
+    unit caster
+    unit teleTarget
+    real duration
+    real lifetime = 0
+    UnitEntity casterEntity
+    CallbackPeriodic pollingLoop
+    IterableMap<int, SortingPosition> sortingPositions
+    real nextSortAngle = 0
+    real sortDistanceOffset = SORT_DISTANCE_INCREMENT
+
+    construct(unit caster, unit teleTarget, real duration)
+        InstantDummyCaster.castTarget(caster.getOwner(), ABILITY_ID_CASTER_BUFF, 1, "curse", caster, caster.getPos() + vec2(100,0))
+        this.caster = caster
+        this.casterEntity = UnitEntity.findForUnit(this.caster)
+        this.duration = duration
+        this.teleTarget = teleTarget
+        this.sortingPositions = new IterableMap<int, SortingPosition>()
+        startBuffPolling()
+
+    function startBuffPolling()
+        this.pollingLoop = doPeriodically(1) cb ->
+            this.lifetime += 1
+            if this.lifetime >= this.duration
+                or not caster.hasAbility(CASTER_BUFF_ID)
+                or not teleTarget.hasAbility(TARGET_BUFF_ID)
+                destroy this
+
+    function canGatherItem(int itemType) returns bool
+        if itemType == 'XRAW' //Raw meat
+            return false
+
+        //Skip herbs if not herb-gatherer
+        if not this.casterEntity instanceof HerbMaster
+            and not this.casterEntity instanceof Omnigatherer
+                if itemType == ITEM_RIVER_ROOT
+                    or itemType == ITEM_RIVER_STEM
+                    or itemType == ITEM_BUTSU
+                    or itemType == ITEM_BLUE_HERB
+                    or itemType == ITEM_ORANGE_HERB
+                    or itemType == ITEM_YELLOW_HERB
+                    or itemType == ITEM_PURPLE_HERB
+                    or itemType == ITEM_MUSHROOM
+                    return false
+
+        return true
+
+    function handleTeleGather(item itm)
+        let itemType = itm.getTypeId()
+
+        if not canGatherItem(itemType)
+            return
+
+        let sortingPos = this.getSortingPos(itemType)
+        this.caster.removeItem(itm)
+        itm.setPos(sortingPos.pos)
+
+    function getSortingPos(int itemType) returns SortingPosition
+        let targetPos = teleTarget.getPos()
+
+        //If a sorting position exists for type, return that
+        if this.sortingPositions.has(itemType)
+            return this.sortingPositions.get(itemType)
+
+        //If a nearby item of the same type is found, set up sorting position there
+        let nearest = findNearestItem(targetPos, sortDistanceOffset * 2.2, i -> i.getTypeId() == itemType)
```

Comment:
Why 2.2?

Reply:
Its supposed to be 10% higher than the radius where the loop tries to find items to place. So that when items might form into big piles and "leak" out of the radius they are still detected as nearby items

Reply:
Added a comment